src/module.c

```diff
@@ -6507,6 +6507,49 @@ const char *RM_CallReplyProto(RedisModuleCallReply *reply, size_t *len) {
     return callReplyGetProto(reply, len);
 }
 
+/* Redis only exposes "side effects" of command execution at the end of an execution unit.
+ *
+ * Side effects currently handled are
+ * - propagating replication/AOF operations
+ * - executing the commands of clients blocked on keys modified by commands
```

Comment:
I would prefer we say, and mentally model, this API like:

```
/* Redis treats multiple invocations of RM_Call() to the same context as if they were executed in the same transaction. Effects
* of these commands are not visible to other clients or visible to the server until the transaction has been completed. This function
* allows completing the transaction without tearing down the context, allowing additional commands to be issued in a new transaction.
```

Reply:
your text seems fine to me.
i would prefer to also list the actual actions (while mentioning that the list may change between versions), i think it makes it easier to understand. but if you're not comfortable with this i'm ok to drop them (people who care or are curious, can easily check the implementation).

Reply:
Yeah, my concern is I don't want to commit to something that may change.

Reply:
i don't want to commit either.. i just thought that listing them makes it easier to understand.
but i'm ok with not listing them.